### ISSUE TITLE
ghost argument for constructor not recognized as specification-context

### ISSUE THREAD

**Issue by tjhance** [dafny-lang/dafny#374]
This code,

```
class C {
  constructor(ghost x: int)
  {
  }
}

function f() : int { 0 }

method main() {
  var c := new C(f());
}
```

wrongly gives an error message,

```
$ daf a.dfy
Dafny 2.3.0.10506
a.dfy(10,17): Error: function calls are allowed only in specification contexts (consider declaring the function a 'function method')
```

Dafny does not seem to recognize that the call `f()` is within a specification context.

**Comment by davidcok** [comment]
Fixed in PR #691.  Added a check on whether the formal argument is ghost; added tests

**Comment by davidcok** [comment]
Added test code from duplicate issue #386 as well
